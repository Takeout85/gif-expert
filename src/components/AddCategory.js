import React from "react";
import propTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = React.useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 1) {
      setCategories((cats) => [inputValue, ...cats]);
    }

    setInputValue("");
  };
  return (
    <form onSubmit={handleSubmit}>
      <div className="searchContainer">
        <input
          className="search"
          type="text"
          name="search"
          onChange={handleInputChange}
          value={inputValue}
          placeholder="Add category"
        />
      </div>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: propTypes.func.isRequired,
};

export default AddCategory;
