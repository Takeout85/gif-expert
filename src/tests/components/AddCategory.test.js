import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import AddCategory from "../../components/AddCategory";

describe("Tests on the < AddCategory /> component", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("should show the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should change the input text", () => {
    const input = wrapper.find("input");
    const value = "Kurisu";

    input.simulate("change", { target: { value } });

    expect(wrapper.find("input").prop("value").trim()).toBe(value);
  });

  test(`should'nt post information with submit`, () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("should  call the setcategories and clean the input text", () => {
    const value = "Kurisu";

    wrapper.find("input").simulate("change", { target: { value } });

    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();

    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
