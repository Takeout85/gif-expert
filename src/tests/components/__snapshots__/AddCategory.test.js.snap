// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on the < AddCategory /> component should show the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="searchContainer"
  >
    <input
      className="search"
      name="search"
      onChange={[Function]}
      placeholder="Add category"
      type="text"
      value=""
    />
  </div>
</form>
`;
